cpp/src/gandiva/precompiled/types.h

```diff
@@ -121,6 +121,7 @@ double months_between_timestamp_timestamp(uint64, uint64);
 int32 mem_compare(const char* left, int32 left_len, const char* right, int32 right_len);
 
 int32 mod_int64_int32(int64 left, int32 right);
```

Comment:
I'm curious to know why this one does not have a context.

Reply:
From the other mod functions (implemented with the `MOD_OP` macro) it looks like expected behavior on divide-by-zero is to return the left operand and set no error: https://github.com/apache/arrow/pull/4820/files#diff-8772261adb65719b136969f075f5f678R54

Reply:
looks like a bug. created ARROW-5882.